src/modal/index.ts

```diff
@@ -0,0 +1,57 @@
+import { alias, tag, utils, Tag } from '@storefront/core';
+
+@alias('modal')
+@tag('gb-modal', require('./index.html'), require('./index.css'))
+class Modal {
+
+  refs: {
+    content: HTMLDivElement,
+  };
+
+  props: Modal.Props = {
+    autoOpen: false,
+  };
+
+  init() {
+    this.set({ visible: this.props.autoOpen });
+  }
+
+  handleOpen = (event: MouseEvent) => {
+    const { body, addEventListener } = utils.WINDOW().document;
+    event.preventDefault();
+    event.stopPropagation();
+    this.set({ visible: true });
+    body.style.overflow = 'hidden';
+    addEventListener('click', this.close);
+  }
+
+  handleClose = (event: MouseEvent) => {
```

Comment:
if we make `updateClose()` an arrow function, can we use it in place of `handleClose()`?